src/gpu_sysman_test.c

```diff
@@ -10,14 +10,21 @@
  *
  * Testing for gpu_sysman.c Sysman API usage and error handling.
  *
- * See:
+ * Sysman API:
  * - https://spec.oneapi.com/level-zero/latest/sysman/PROG.html
  * - https://spec.oneapi.io/level-zero/latest/sysman/api.html
  *
- * Building unit-tests:
- *   gcc -I. -Idaemon  -I/path/to/level-zero -O3 -g --coverage -Werror \
- *       -Wall -Wextra -Wpedantic -Wcast-align=strict -Wformat-security \
- *       gpu_sysman_test.c -o test_plugin_gpu_sysman
+ * Installing build dependencies:
+ * - Fedora: sudo dnf install oneapi-level-zero-devel
+ * - Debian: sudo apt install libze-dev
+ *
+ * Building unit-tests (after running "configure", so that
+ * "config.h" + other generated includes are present):
+ *	gcc -I. -Idaemon -DHAVE_CONFIG_H -O3 -g --coverage -Werror \
+ *	 -Wall -Wextra -Wpedantic -Wcast-align=strict -Wformat-security \
+ *	 -Wnull-dereference -Wstrict-overflow=2 -Warray-bounds=2 \
+ *	 -D_FORTIFY_SOURCE=2 -Wno-aggressive-loop-optimizations \
+ *	 gpu_sysman_test.c -o test_plugin_gpu_sysman
```

Comment:
Why do you need to document this?

The tests should automatically be built and run as part of `make check`. If you need to build this test individuall, run `make test_plugin_gpu_sysman`. If you want to run only this test, you can use `make test_plugin_gpu_sysman.log` (only really useful during development).

Reply:
> Why do you need to document this?

* Comment documents everything that I'm testing with that file (used compiler options are just one of those items)
* It lists (stricter) compiler options that included collectd core files should IMHO pass (reason for the previous fix commits to them)

> The tests should automatically be built and run as part of `make check`. If you need to build this test individually, run `make test_plugin_gpu_sysman`. If you want to run only this test, you can use `make test_plugin_gpu_sysman.log` (only really useful during development).

Thanks!

Often I will (build and) run it separately though, because I want that to be done under Valgrind (maybe later compile tests with ASAN too), and I'm interested also in test coverage:
```
Optimized build coverage info:

File 'gpu_sysman_test.c'
Lines executed:86.50% of 511

File 'gpu_sysman.c'
Lines executed:87.46% of 1132

File 'daemon/metric.c'
Lines executed:27.10% of 369

File 'utils/strbuf/strbuf.c'
Lines executed:0.00% of 126

File 'utils/utf8/utf8.c'
Lines executed:71.43% of 7

File 'utils/metadata/meta_data.c'
Lines executed:1.29% of 309

File 'utils/common/common.c'
Lines executed:3.41% of 499

File 'daemon/utils_cache_mock.c'
Lines executed:0.00% of 19
```

(As you can see, some of the includes are needed just duo to another collectd core file referring symbols in them, nothing is actually used from them during Sysman plugin tests.)

Reply:
Documented this a bit better in the comment.